src/megaclient.cpp

```diff
@@ -1825,9 +1787,10 @@ void MegaClient::exec()
                                 if (((*it)->state == SYNC_ACTIVE || (*it)->state == SYNC_INITIALSCAN)
                                  && !(*it)->dirnotify->notifyq[DirNotify::DIREVENTS].size()
                                  && !(*it)->dirnotify->notifyq[DirNotify::RETRY].size()
-                                 && !syncadding)
+                                 && !syncadding && syncuprequired && !syncnagleretry)
```

Comment:
`syncuprequired` is per sync, but more than one sync might require a `syncup`. Hence, the flag should be per sync, maybe part of the `Sync` object. The global checkup would be a OR of every single flag.
Note that this issue could be affecting the `syncdownrequired` on the same way.